### PR TITLE
vz 10274 cli vpo already deployed

### PR DESCRIPTION
**Pull request by AbeHern**
Added changes to introduce a new --skip-platform-operator flag

### CODE REVIEW

File: tools/vz/pkg/constants/constants.go [hunk 1]
```diff
@@ -14,50 +14,52 @@ const GlobalFlagContext = "context"
...
+	LogFormatFlag                = "log-format"
+	LogFormatHelp                = "The format of the log output. Valid output formats are \"simple\" and \"json\"."
+	FilenameFlag                 = "filename"
+	FilenameFlagShorthand        = "f"
+	FilenameFlagHelp             = "Path to file containing Verrazzano custom resource.  This flag can be specified multiple times to overlay multiple files.  Specifying \"-\" as the filename accepts input from stdin."
+	SkipConfirmationFlagHelp     = "Non-interactive mode - assumes the answers to all interactive questions to be 'Y'."
+	SkipConfirmationFlag         = "skip-confirmation"
+	SkipConfirmationShort        = "y"
+	SkipPlatformOperatorFlag     = "skip-platform-operator"
+	SkipPlatformOperatorFlagHelp = "Installing the operator is skipped, but continues on to install Verrazzano."
```
Comment (michael-cico) [inline tools/vz/pkg/constants/constants.go:53]: I think `operator` is supposed to be `platform operator` or `Verrazzano Platform Operator` right?

Comment (AbeHern) [reply tools/vz/pkg/constants/constants.go:53]: You are correct, it should be platform operator

File: tools/vz/cmd/install/install.go [hunk 2]
```diff
@@ -454,3 +481,43 @@ func ValidateCR(cmd *cobra.Command, obj *unstructured.Unstructured, vzHelper hel
...
+	return true, nil
+}
+
+func continueReapply(skipPlatformOperator bool) (bool, error) {
+	if skipPlatformOperator {
+		return false, nil
+	}
+	var response string
+	scanner := bufio.NewScanner(os.Stdin)
+	fmt.Print("Do you want to reinstall the Verrazzano Platform Operator? [y/N]: ")
```
Comment (bobdonat) [inline tools/vz/cmd/install/install.go:512]: this needs to take into consideration the --skip-confirmation flag

File: tools/vz/cmd/install/install.go [hunk 3]
```diff
@@ -248,10 +259,13 @@ func runCmdInstall(cmd *cobra.Command, args []string, vzHelper helpers.VZHelper)
 }
 
 func installVerrazzano(cmd *cobra.Command, vzHelper helpers.VZHelper, vz clipkg.Object, client clipkg.Client, version string, vpoTimeout time.Duration, obj *unstructured.Unstructured) error {
-	// Wait for the platform operator to be ready before we create the Verrazzano resource.
-	_, err := cmdhelpers.WaitForPlatformOperator(client, vzHelper, v1beta1.CondInstallComplete, vpoTimeout)
-	if err != nil {
-		return err
+	// Do not wait for platform operator if skip-platform-operator flag is enabled
+	if !cmd.PersistentFlags().Changed(constants.SkipPlatformOperatorFlag) {
```
Comment (bobdonat) [inline tools/vz/cmd/install/install.go:263]: If you say N to prompt to skip the install of VPO the message "Waiting for verrazzano-platform-operator to be ready before starting install - 1 seconds" will still be displayed.

**Review by michael-cico — APPROVED** [review]

File: tools/vz/cmd/install/install.go [hunk 4]
```diff
@@ -216,16 +219,24 @@ func runCmdInstall(cmd *cobra.Command, args []string, vzHelper helpers.VZHelper)
 			return err
 		}
 
-		// Delete leftover verrazzano-platform-operator deployments after an abort.
-		// This allows for the verrazzano-platform-operator validatingWebhookConfiguration to be updated with the correct caBundle.
-		err = cmdhelpers.DeleteFunc(client)
+		// finds a running Verrazzano Platform Operator and determines whether to suppress the reinstall prompt
```
Comment (bobdonat) [inline tools/vz/cmd/install/install.go:222]: i think this comment is wrong

File: tools/vz/cmd/install/install.go (hunk 3 above)
Comment (bobdonat) [reply tools/vz/cmd/install/install.go:263]: I think you want to check the flag returned from reapplyPlatformOperator instead

File: tools/vz/cmd/install/install.go (hunk 4 above)
Comment (AbeHern) [reply tools/vz/cmd/install/install.go]: Changed

File: tools/vz/cmd/install/install.go (hunk 3 above)
Comment (AbeHern) [reply tools/vz/cmd/install/install.go]: Changed

File: tools/vz/cmd/install/install.go (hunk 2 above)
Comment (AbeHern) [reply tools/vz/cmd/install/install.go:516]: Used the --skip-confirmation flag into the existing logic to restart the VPO

**Review by bobdonat — APPROVED** [review]

**Pull request merged**